pkg/controller/apiserver/handlers/endpoint/handler.go

```diff
@@ -32,10 +32,10 @@ func HandleFunc(eq networkpolicy.EndpointQuerier) http.HandlerFunc {
 			return
 		}
 		// query endpoint and handle response errors
-		endpointQueryResponse := eq.QueryNetworkPolicies(namespace, podName)
-		if endpointQueryResponse.Error != nil {
+		endpointQueryResponse, err := eq.QueryNetworkPolicies(namespace, podName)
+		if err != nil {
 			//TODO: need to address errors here which are not resource not found errors
-			http.Error(w, endpointQueryResponse.Error.Error(), http.StatusNotFound)
+			http.Error(w, err.Error(), http.StatusNotFound)
```

Comment:
I think I suggested looking at whether the `Endpoints` field was `nil` to distinguish between NotFound and InternalError? And returning `nil` in `QueryNetworkPolicies` if the only "error" is that the Pod wasn't found.

Reply:
```
if err == nil && endpointQueryResponse.Endpoints == nil {
    // return not found
} else if err != nil {
    // return internal error
}
```